java/ql/src/Telemetry/AutomodelApplicationModeCharacteristics.qll

```diff
@@ -67,19 +67,19 @@ module ApplicationCandidatesImpl implements SharedCharacteristics::CandidateSig
 
   predicate isKnownKind = AutomodelJavaUtil::isKnownKind/2;
 
-  predicate isSink(Endpoint e, string kind) {
+  predicate isSink(Endpoint e, string kind, string provenance) {
     exists(string package, string type, string name, string signature, string ext, string input |
       sinkSpec(e, package, type, name, signature, ext, input) and
-      ExternalFlow::sinkModel(package, type, _, name, [signature, ""], ext, input, kind, _)
+      ExternalFlow::sinkModel(package, type, _, name, [signature, ""], ext, input, kind, provenance)
     )
     or
-    isCustomSink(e, kind)
+    isCustomSink(e, kind) and provenance = "custom-sink"
```

Comment:
Unrelated, but happy to tell you that `ArgumentToExec` and `RequestForgerySink` shouldn't need special treatment as custom sinks anymore :)

Reply:
awesome, I'll remove the special treatment as a drive-by change in this PR, ok?

Reply:
[2f2f507](https://github.com/github/codeql/pull/13747/commits/2f2f507a5da5b460c426810f076ef9ffe1dc8d6a)